Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,485 @@
+from .Compiler import Version
+from .Compiler.Nodes import *
+from .Compiler.ExprNodes import * 
+from .Compiler.ModuleNode import ModuleNode
+from .CodeWriter import DeclarationWriter
+from .Compiler.ParseTreeTransforms import CythonTransform
+import cython 
+from .Compiler.Main import Context
+from .Utils import open_new_file
+
+cython.declare(PyrexTypes=object, Naming=object, ExprNodes=object, Nodes=object,
+               Options=object, UtilNodes=object, LetNode=object,
+               LetRefNode=object, TreeFragment=object, EncodedString=object,
+               error=object, warning=object, copy=object, _unicode=object)
+
+
+# Inspired by and based around https://github.com/cython/cython/pull/3818
+# with some less lazy changes to it and a few minor improvements and optimzations...
+
+# Decided to revert to an older variant I had wrote of this code for the sake of 
+# maintainability - Vizonex
+
+
+
+# TODO Save this implemenation commented out if required....
+# if sys.version_info >= (3, 9):
+#     typing_module = "typing"
+# else:
+#     typing_module = "typing_extensions"
+
+def ctype_name(arg, node:"Node") -> str:
+
+    # TODO Make a better conversion function...
+    if arg.type and hasattr(arg.type, "name"):
+        # Used C declared type...
+        # TODO see about using a check to see if users wants to include cython's shadow varaibales...
+        return arg.type.name
+        
+    py_name = node.type.return_type.py_type_name() # type: ignore
+    if "(int, long)" == py_name:
+        return "int"
+    
+    return py_name
+
+
+def translate_annotations(node) -> list[str]:
```

Comment:
We also can't assume that we run on a version of Python that supports type annotations